nemo/core/neural_types/nmtensor_registry.py

```diff
@@ -0,0 +1,88 @@
+# =============================================================================
+# Copyright (c) 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+
+class NmTensorNameRegistry:
```

Comment:
So any particular reason why those shouldn't be weak references?

Reply:
if you store names only - not important anymore

Reply:
See https://github.com/NVIDIA/NeMo/pull/615#discussion_r431406913